src/vmspawn/vmspawn.c

```diff
@@ -1861,6 +1861,17 @@ static int run_virtual_machine(int kvm_device_fd, int vhost_device_fd) {
                 r = unit_name_to_prefix(trans_scope, &scope_prefix);
                 if (r < 0)
                         return log_error_errno(r, "Failed to strip .scope suffix from scope: %m");
+
+                /* on distros that provide their own sshd@.service file we need to provide a dropin which
+                 * picks up our public key credential */
+                r = machine_credential_set(&arg_credentials,
+                                "systemd.unit-dropin.sshd-vsock@.service:"
+                                "[Service]\n"
+                                "ExecStart=\n"
+                                "ExecStart=/usr/sbin/sshd -i -o 'AuthorizedKeysFile=${CREDENTIALS_DIRECTORY}/ssh.ephemeral-authorized_keys-all .ssh/authorized_keys'\n"
+                                "ImportCredential=ssh.ephemeral-authorized_keys-all\n");
+                if (r < 0)
+                        return log_error_errno(r, "Failed to set credential systemd.unit-dropin.sshd-vsock@.service: %m");
```

Comment:
hmm, i wonder if we can solve this instead via an sshd_config.d/ drop-in that we ship?

Reply:
ie. src/userdb/20-systemd-userdb.conf.in uses this already. maybe we can just extend things like that?

Reply:
I'm not sure we can, what happens if it is an encrypted credential? we can't just set AuthorizedKeysFile to AuthorizedKeysFile=/run/credentials/@system/ssh.ephemeral-authorized_keys-all can we?

I just tried with the following snippet:
```
Match Host vsock/*
        AuthorizedKeysFile=/run/credentials/@system/ssh.ephemeral-authorized_keys-all .ssh/authorized_keys
```
and the drop in removed and it didn't work.
I'm fairly sure this is the correct Match and cat-ing that path does get the right value.